packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+   *
+   * @default no additional behaviors are added.
+   */
+  readonly additionalBehaviors?: Record<string, BehaviorOptions>;
+
+  /**
+   * A certificate to associate with the distribution. The certificate must be located in N. Virginia (us-east-1).
+   *
+   * @default the CloudFront wildcard certificate (*.cloudfront.net) will be used.
+   */
+  readonly certificate?: acm.ICertificate;
+
+  /**
+   * The price class that corresponds with the maximum price that you want to pay for CloudFront service.
+   * If you specify PriceClass_All, CloudFront responds to requests for your objects from all CloudFront edge locations.
+   * If you specify a price class other than PriceClass_All, CloudFront serves your objects from the CloudFront edge location
+   * that has the lowest latency among the edge locations in your price class.
+   *
+   * @default PRICE_CLASS_ALL
+   */
+  readonly priceClass?: PriceClass;
+
+  /**
+   * How CloudFront should handle requests that are not successful (e.g., PageNotFound).
+   *
+   * @default - No custom error configuration.
+   */
+  readonly errorConfigurations?: CustomErrorResponse[];
+}
+
+/**
+ * A CloudFront distribution with associated origin(s) and caching behavior(s).
+ *
+ * @experimental
+ */
+export class Distribution extends Resource implements IDistribution {
+
+  /**
+   * Creates a Distribution construct that represents an external (imported) distribution.
+   */
+  public static fromDistributionAttributes(scope: Construct, id: string, attrs: DistributionAttributes): IDistribution {
+    return new class extends Resource implements IDistribution {
+      public readonly domainName: string;
+      public readonly distributionId: string;
+
+      constructor() {
+        super(scope, id);
+        this.domainName = attrs.domainName;
+        this.distributionId = attrs.distributionId;
+      }
+    }();
+  }
+
+  public readonly domainName: string;
+  public readonly distributionId: string;
+
+  private readonly defaultBehavior: CacheBehavior;
+  private readonly additionalBehaviors: CacheBehavior[] = [];
+  private readonly origins: Set<Origin> = new Set<Origin>();
+
+  private readonly errorConfigurations: CustomErrorResponse[];
+  private readonly certificate?: acm.ICertificate;
+
+  constructor(scope: Construct, id: string, props: DistributionProps) {
+    super(scope, id);
+
+    if (props.certificate) {
+      const certificateRegion = Stack.of(this).parseArn(props.certificate.certificateArn).region;
+      if (!Token.isUnresolved(certificateRegion) && certificateRegion !== 'us-east-1') {
+        throw new Error('Distribution certificates must be in the us-east-1 region.');
+      }
+    }
+
+    this.defaultBehavior = new CacheBehavior({ pathPattern: '*', ...props.defaultBehavior });
+    this.addOrigin(this.defaultBehavior.origin);
+    if (props.additionalBehaviors) {
+      Object.entries(props.additionalBehaviors).forEach(([pathPattern, behaviorOptions]) => {
```

Comment:
nit: Prefer normal for-loops instead of `.forEach()` to ease refactoring (for example, if you want to break from a .forEach lambda, you need `return`. Then, if you move this code outside the lambda, this return is going to bite).